lib/elixir/lib/kernel/lexical_tracker.ex

```diff
@@ -171,8 +172,21 @@ defmodule Kernel.LexicalTracker do
   defp add_compile(d, module, :runtime), do: :ets.insert_new(d, {{:mode, module}, :runtime})
   defp add_compile(d, module, :compile), do: :ets.insert(d, {{:mode, module}, :compile})
 
-  defp add_directive(d, module, line, warn, tag) do
+  defp add_directive(d, m_or_mfa, line, warn, :import = tag) when is_atom(m_or_mfa) do
+    # remove possible previous `{m, f, a}` entries, because here the entire module
+    # has been imported instead of selected functions.
+    module = m_or_mfa
```

Comment:
Move this to logic to `handle_cast({:add_import, ...})` because we then can at least always call add_directive at the end.